libcudacxx/test/libcudacxx/std/containers/sequences/inplace_vector/assign.pass.cpp

```diff
@@ -0,0 +1,340 @@
+  { // inplace_vector<T, N>::assign(iter, iter), with input iterators shrink to empty range
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 0> expected = {};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N>::assign(iter, iter), with input iterators shrinking
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(!vec.empty());
+    assert(equal_range(vec, expected));
+  }
+
+  { // inplace_vector<T, N>::assign(iter, iter), with input iterators growing
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 6> expected = {T(42), T(1), T(42), T(1337), T(0), T(42)};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(!vec.empty());
+    assert(equal_range(vec, expected));
+  }
+
+  { // inplace_vector<T, N>::assign(iter, iter), with forward iterators empty range
+    const cuda::std::array<T, 0> expected = {};
+    inplace_vector vec{};
+    vec.assign(expected.begin(), expected.end());
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N>::assign(iter, iter), with forward iterators shrinking to empty
+    const cuda::std::array<T, 0> expected = {};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec.assign(expected.begin(), expected.end());
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N>::assign(iter, iter), with forward iterators shrinking
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec.assign(expected.begin(), expected.end());
+    assert(!vec.empty());
+    assert(equal_range(vec, expected));
+  }
+
+  { // inplace_vector<T, N>::assign(iter, iter), with forward iterators growing
+    const cuda::std::array<T, 6> expected = {T(42), T(1), T(42), T(1337), T(0), T(42)};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec.assign(expected.begin(), expected.end());
+    assert(!vec.empty());
+    assert(equal_range(vec, expected));
+  }
+
+  { // inplace_vector<T, N>::assign(initializer_list), empty range
+    const cuda::std::initializer_list<T> expected = {};
+    inplace_vector vec{};
+    vec.assign(expected);
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N>::assign(initializer_list), shrinking to empty
+    const cuda::std::initializer_list<T> expected = {};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec.assign(expected);
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N>::assign(initializer_list), shrinking
+    const cuda::std::initializer_list<T> expected = {T(42), T(42)};
```

Comment:
I attempted to reduce the bug and reported [DevCom-10720847](https://developercommunity.visualstudio.com/t/10720847). Should we temporarily suppress the warning for MSVC?

Reply:
The unfortunate issue is that we cannot suppress the warning. I believe we will need to completely #ifdef out those tests